src/components/Modals/ReplaceCourseModal.vue

```diff
@@ -0,0 +1,291 @@
+      // opens the add modal if the course does not exist
+      if (count === 0) {
+        this.needToAdd = true;
+        // opens the duplicates modal if the course exists 2+ times
+      } else if (count > 1) {
+        this.hasDuplicates = true;
+        // closes the modal if the course exists exactly once
+      } else {
+        this.closeCurrentModal();
+      }
+    },
+    selectCourse(result: CornellCourseRosterCourse) {
+      this.selectedCourse = result;
+      this.$emit('select-course', this.selectedCourse);
+      this.getReqsRelatedToCourse(result);
+      this.handleAdd();
+    },
+    closeCurrentModal() {
+      this.$emit('close-replace-course-modal');
+    },
+    getReqsRelatedToCourse(selectedCourse: CornellCourseRosterCourse) {
+      const {
+        relatedRequirements,
+        selfCheckRequirements,
+        automaticallyFulfilledRequirements,
+      } = getRelatedUnfulfilledRequirements(
+        selectedCourse,
+        store.state.groupedRequirementFulfillmentReport,
+        store.state.onboardingData,
+        store.state.toggleableRequirementChoices,
+        store.state.overriddenFulfillmentChoices,
+        store.state.userRequirementsMap
+      );
+      const automaticallyFulfilledRequirementIds = new Set(
+        automaticallyFulfilledRequirements.map(({ id }) => id)
+      );
+
+      this.automaticallyFulfilledRequirements = automaticallyFulfilledRequirements.map(
+        ({ name }) => name
+      );
+      this.relatedRequirements = relatedRequirements.filter(
+        req => !automaticallyFulfilledRequirementIds.has(req.id)
+      );
+      this.selfCheckRequirements = selfCheckRequirements.filter(
+        req => !automaticallyFulfilledRequirementIds.has(req.id)
+      );
+      if (this.relatedRequirements.length > 0) {
+        this.selectedRequirementID = this.relatedRequirements[0].id;
+      } else {
+        this.selectedRequirementID = '';
+      }
+    },
+    addItem() {
+      if (this.editMode) {
+        this.editMode = false;
+      } else {
+        this.addCourse();
+      }
+    },
+    addCourse() {
+      if (this.selectedCourse == null) return;
+      this.$emit('add-course', this.selectedCourse, {
+        optOut: getRelatedRequirementIdsForCourseOptOut(
+          this.selectedCourse.crseId,
+          this.selectedRequirementID,
+          store.state.groupedRequirementFulfillmentReport,
+          store.state.toggleableRequirementChoices,
+          store.state.userRequirementsMap
+        ),
+        // Only include the selected requirement from opt-in.
+        acknowledgedCheckerWarningOptIn: this.selfCheckRequirements
+          .filter(it => it.id === this.selectedRequirementID)
+          .map(it => it.id),
+        arbitraryOptIn: {},
+      });
+      this.closeCurrentModal();
+    },
+    onSelectedChange(selected: string) {
+      this.selectedRequirementID = selected;
+    },
+    backOrCancel() {
+      if (this.leftButtonText === 'Back') {
```

Comment:
I'm not too keen on using the text to display to determine which state we're in; I think it should be the other way around. For example, we could have a `State` enum that could be used to determine the text to display in each button.